### PR TITLE
Add 'project' Makefile target

### PR DESCRIPTION
**Pull request by eriknyquist**
This target creates a new empty project. Reqested by multiple bootcamp
participants.

@calvinatintel @bbaltz505 @SidLeung @bigdinotech Please review.

Note that we will do usability testing of this today at bootcamp; if it works well then I'll add it to other trees. Please don't merge until it is explicitly requested

### CODE REVIEW

**Comment by bbaltz505** [comment]
Barebones project looks good. Just curious how it fits in to the current compile/upload targets (X86_PROJ_DIR and ARC_PROJ_DIR params)

File: Makefile [hunk 1]
```diff
@@ -137,3 +139,15 @@ debug-x86:
 
 debug-arc:
 	$(CODK_ARC_DIR)/arc32/bin/arc-elf32-gdb $(ARC_PROJ_DIR)/arc-debug.elf
+
+project:
+	@if [ -d $(PROJNAME) ]; then echo "$(PROJNAME) already exists."; exit 1; fi
+	mkdir $(PROJNAME)
+	mkdir $(PROJNAME)/arc
+	mkdir $(PROJNAME)/x86
+	mkdir $(PROJNAME)/x86/src
```
Comment (bbaltz505) [inline Makefile]: You could just do:
mkdir -p  $(PROJNAME)/x86/src
mkdir -p  $(PROJNAME)/arc

Comment (bbaltz505) [inline Makefile]: or... just:
cp -r $(PROJ_TEMPLATE_DIR) $(PROJNAME)
if the template directory has all of the bare-bones files

**Comment by SidLeung** [comment]
Empty project for both ARC and i86?

**Comment by eriknyquist** [comment]
@SidLeung yes.

@bbaltz505 Yes, maintaining the complete directory structure and just copying it is a better solution.

**Comment by eriknyquist** [comment]
Simplified as per suggestion from @bbaltz505

**Comment by bbaltz505** [comment]
I copied Blink.ino.cpp into a project/arc folder and ran:
    make upload-arc-dfu ARC_PROJ_DIR=$(pwd)/project/arc
and it built and uploaded just fine.

**Comment by eriknyquist** [comment]
If I could figure out a way to have the 'project' target set ARC/X86_PROJ_DIR to the newly created dirs and export them to parent shell..... that would be great. I'll look at it tomorrow.

**Comment by calvinatintel** [comment]
I'm pretty sure you can't set env vars of the shell from a Makefile...

**Comment by eriknyquist** [comment]
superseded by #27

**Pull request closed without merging**